core/src/main/java/org/elasticsearch/ingest/core/Pipeline.java

```diff
@@ -93,25 +96,26 @@ public Pipeline create(String id, Map<String, Object> config, Map<String, Proces
             return new Pipeline(id, description, compoundProcessor);
         }
 
-        private List<Processor> readProcessors(String fieldName, Map<String, Processor.Factory> processorRegistry, Map<String, Object> config) throws Exception {
-            List<Map<String, Map<String, Object>>> onFailureProcessorConfigs = ConfigurationUtils.readOptionalList(config, fieldName);
-            List<Processor> onFailureProcessors = new ArrayList<>();
-            if (onFailureProcessorConfigs != null) {
-                for (Map<String, Map<String, Object>> processorConfigWithKey : onFailureProcessorConfigs) {
+        private List<Processor> readProcessors(AtomicInteger processorIdGenerator, String fieldName, Map<String, Processor.Factory> processorRegistry, Map<String, Object> config) throws Exception {
+            List<Map<String, Map<String, Object>>> processorConfigs = ConfigurationUtils.readOptionalList(config, fieldName);
+            List<Processor> processors = new ArrayList<>();
+            if (processorConfigs != null) {
+                for (Map<String, Map<String, Object>> processorConfigWithKey : processorConfigs) {
                     for (Map.Entry<String, Map<String, Object>> entry : processorConfigWithKey.entrySet()) {
-                        onFailureProcessors.add(readProcessor(processorRegistry, entry.getKey(), entry.getValue()));
+                        processors.add(readProcessor(processorIdGenerator, processorRegistry, entry.getKey(), entry.getValue()));
                     }
                 }
             }
 
-            return onFailureProcessors;
+            return processors;
         }
 
-        private Processor readProcessor(Map<String, Processor.Factory> processorRegistry, String type, Map<String, Object> config) throws Exception {
+        private Processor readProcessor(AtomicInteger processorIdGenerator, Map<String, Processor.Factory> processorRegistry, String type, Map<String, Object> config) throws Exception {
             Processor.Factory factory = processorRegistry.get(type);
             if (factory != null) {
-                List<Processor> onFailureProcessors = readProcessors(ON_FAILURE_KEY, processorRegistry, config);
-                Processor processor = factory.create(config);
+                String processorId = ConfigurationUtils.readStringProperty(config, PROCESSOR_ID_KEY, "processor_" + processorIdGenerator.getAndIncrement());
```

Comment:
the processor ids have no meaning on our side and are completely meta. So its fine. It is more of tag then it is an id, so others that are integrating with ingest.